### ISSUE TITLE
comparison_chain false positive when every test is the same comparison to a number

### ISSUE THREAD

**Issue by csnover** [rust-lang/rust-clippy#5212]
Test case:

```rust
fn mutate_something() -> u8 {
  1
}

fn main() {
  let _ = if mutate_something() == 0 {
    1
  } else if mutate_something() == 0 {
    2
  } else {
    3
  };
}
```

If any of the tests are different then `comparison_chain` doesn’t trigger.

[Here is a reference to the actual code](https://github.com/csnover/earthquake-rust/blob/c48b39872c6725afee78dfbf3c641752834e84e6/src/compression.rs#L182-L209), in case you wonder why someone would do such a thing :-)

clippy 0.0.212 (69f99e74 2019-12-14)

**Comment by flip1995** [comment]
I guess, we should not lint, if only one variant of `Ordering` is used. In this case `Ordering::Equal`.
